pkg/agent/cniserver/ipam/antrea_ipam_controller.go

```diff
@@ -91,34 +117,100 @@ func (c *AntreaIPAMController) Run(stopCh <-chan struct{}) {
 	}()
 
 	klog.InfoS("Starting", "controller", controllerName)
-	if !cache.WaitForNamedCacheSync(controllerName, stopCh, c.namespaceInformer.Informer().HasSynced, c.ipPoolInformer.Informer().HasSynced) {
+	if !cache.WaitForNamedCacheSync(controllerName, stopCh, c.namespaceInformer.Informer().HasSynced, c.ipPoolInformer.Informer().HasSynced, c.podInformer.HasSynced) {
 		return
 	}
 
 	<-stopCh
 }
 
-func (c *AntreaIPAMController) getIPPoolsByNamespace(namespace string) []string {
+func (c *AntreaIPAMController) getIPPoolsByPod(namespace, name string) ([]string, []net.IP, *crdv1a2.IPAddressOwner, error) {
+	// Find IPPool by Pod
+	var IPs []net.IP
+	var reservedOwner *crdv1a2.IPAddressOwner
+	pod, err := c.podLister.Pods(namespace).Get(name)
+	if err != nil {
+		// When get Pod failed, try to get information from allocated IPs of all IPPools.
```

Comment:
@annakhm This part will affect the logic for CNI DEL. Please take a look.

Reply:
When this can happen? Why we try getting information from IPPools? Could you add more comments here?

Reply:
Added comments.
For CNI DEL case, getting Pod may fail. Try to get information from allocated IPs of all IPPools.